cli/talekclient/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/privacylab/talek/common"
+	"github.com/privacylab/talek/libtalek"
+)
+
+var config = flag.String("config", "talek.conf", "Client configuration for talek")
+var create = flag.Bool("create", false, "Create a new talek handle")
+var share = flag.String("share", "", "Create a read-only version of the topic for sharing")
+var handle = flag.String("topic", "talek.handle", "The talek handle to use")
+var write = flag.String("write", "", "A message to append to the log (If not specified, the next item will be read.)")
+var read = flag.Bool("read", false, "Read from the provided topic")
+var verbose = flag.Bool("verbose", false, "Print diagnostic information")
+
+// The CLI client will read or write a single item for talek
+func main() {
+	flag.Parse()
+
+	// Config
+	config := libtalek.ClientConfigFromFile(*config)
+	if config.Config == nil && *verbose {
+		fmt.Fprintln(os.Stderr, "Common configuration will be fetched from frontend.")
+	}
+
+	topicdata, err := ioutil.ReadFile(*handle)
+	if err != nil {
+		panic(err)
+	}
+	topic := libtalek.Topic{}
+	if *create {
+		nt, newerr := libtalek.NewTopic()
+		if newerr != nil {
+			panic(err)
+		}
+		topic = *nt
+	} else {
+		if err = json.Unmarshal(topicdata, &topic); err != nil {
+			panic(err)
+		}
+	}
+
+	if len(*share) > 0 {
+		handle := topic.Handle
+		handleBytes, handleerr := json.Marshal(handle)
+		if handleerr != nil {
+			panic(handleerr)
+		}
+		ioutil.WriteFile(*share, handleBytes, 0640)
+		fmt.Fprintf(os.Stderr, "Read-only topic written to %s\n", *share)
+	}
+
+	// Client-connected activity below.
+	leaderRPC := common.NewFrontendRPC("RPC", config.FrontendAddr)
+
+	client := libtalek.NewClient("Client", *config, leaderRPC)
+	if client == nil {
+		panic("could not create talek client")
+	} else if *verbose {
+		fmt.Fprintln(os.Stderr, "Connection to RPC established.")
+	}
+
+	if *read == false && len(*write) > 0 {
```

Comment:
That's to handle if neither were specified. My question was the case when both are specified. Currently it'll just read and ignore the write silently